data-runtime/src/main/kotlin/io/micronaut/data/runtime/criteria/KCriteriaBuilderExt.kt

```diff
@@ -41,6 +42,24 @@ fun <E, I, K : Collection<I>?> From<*, E>.joinMany(prop: KProperty1<out E, K>, j
     }
 }
 
+@Experimental
+fun <E, I, K : List<I>?> From<*, E>.joinMany(prop: KProperty1<out E, K>, joinType: JoinType? = null): ListJoin<E, I> {
+    return if (joinType == null) {
+        this.joinList(prop.name)
+    } else {
+        this.joinList(prop.name, joinType)
+    }
+}
```

Comment:
I think it would be interesting to return an instance of `ListJoin` when the joinMany is done on a `List`. This gives the developers more control on what they want to do with its results. Same thing for `SetJoin`.

Reply:
Same question as before how is the compiler going to decide on the method

Reply:
The compiler will check if the Upper Bound of K is a List, and will decide based on that to choose this method. Same thing for Set, or any other Collection type.